src/pykiso/lib/auxiliaries/ykush_auxiliary.py

```diff
@@ -0,0 +1,409 @@
+        """Get a list of all the states of the ports.
+
+        :raises YkushStatePortNotRetrieved: The state couldn't be retrieved
+        :return: list with ON, OFF
+        """
+        return [self.get_str_state(state) for state in self.get_all_ports_state()]
+
+    def get_firmware_version(self) -> Tuple[int]:
+        """Returns a tuple with YKUSH firmware version in format (major, minor)."""
+
+        status, major, minor = self._raw_sendreceive([0xF0])[:3]
+        if status == YKUSH_PROTO_OK_STATUS:
+            self._firmware_major_version, self._firmware_minor_version = (
+                major,
+                minor,
+            )
+        else:
+            # early devices will not recognize it, figure it out from serial
+            self._firmware_major_version = 1
+            self._firmware_minor_version = (
+                2
+                if "YK2" in self.get_serial_number_string()
+                else 255
+                if "YKD2" in self.get_serial_number_string()
+                else 0
+            )
+        return self._firmware_major_version, self._firmware_minor_version
+
+    def set_port_state(self, port_number: int, state: int):
+        """Set a specific port On or Off.
+
+        :raises YkushSetStateError: if the operation had an error
+        :param state: 1 (port on), 0 (port off)
+        """
+        self.check_port_number(port_number)
+        self._raw_sendreceive(
+            [(state == YKUSH_PORT_STATE_ON and 0x10 or 0x0) | port_number]
+        )
+
+        try:
+            state_port = self.get_port_state(port_number)
+        except YkushStatePortNotRetrieved as e:
+            raise YkushSetStateError(
+                f"The state of the action to power {self.get_str_state(state)}"
+                "couldn't be confirmed because the state of the port can't be retrieved"
+            ) from e
+        if state_port != state:
+            raise YkushSetStateError(
+                "There was an error trying to power "
+                f"{self.get_str_state(state).lower()} the port,"
+                f"the port {port_number} is {self.get_str_state(state_port)}"
+            )
+
+    def set_port_on(self, port_number: int):
+        """Power on a specific port.
+
+        :raises YkushSetStateError: if the operation had an error
+        """
+        self.set_port_state(port_number, state=YKUSH_PORT_STATE_ON)
+
+    def set_port_off(self, port_number: int):
+        """Power off a specific port.
+
+        :raises YkushSetStateError: if the operation had an error
+        """
+        self.set_port_state(port_number, state=YKUSH_PORT_STATE_OFF)
+
+    def set_all_ports(self, state: int):
+        """Power off or on all YKUSH ports.
+
+        :param state: state wanted 1 for On, 0 for Off
+        :raises YkushSetStateError: if the operation had an error
+        """
+        self._raw_sendreceive([(state == YKUSH_PORT_STATE_ON and 0x1A or 0x0A)])
```

Comment:
The more I read this line, the less I understand it 😵

It's very hard to read and understand like that IMO, at least split it on another previous line and add comments 

Also, when performing bitwise ops on integers I prefer to use & and | (allows to emphasize the difference with conditionals)

Reply:
Finally got it!

Please replace this with `0x1A if state == YKUSH_PORT_STATE_ON else 0x0A` for the sake of readbility

Reply:
Done